src/coreclr/jit/earlyprop.cpp

```diff
@@ -388,12 +388,13 @@ GenTree* Compiler::optEarlyPropRewriteTree(GenTree* tree, LocalNumberToNullCheck
                     if ((checkConstVal >= 0) && (checkConstVal < actualConstVal))
                     {
                         GenTree* comma = check->gtGetParent(nullptr);
-                        if ((comma != nullptr) && comma->OperIs(GT_COMMA) && (comma->gtGetOp1() == check))
+                        if (((comma != nullptr) && comma->OperIs(GT_COMMA) && (comma->gtGetOp1() == check)) ||
```

Comment:
Seems like this could handle the parentless case (top level check) too?

Reply:
I have discovered that this line is trickier than it looks, as it seems like `gtGetParent` will just AV if it cannot find the parent. Will try to come up with a test that exercises this code path and look into fixing `gtGetParent` so that it returns `nullptr` if no parent is found (which should be equivalent to argument being a root, at least before rationalization?).

Reply:
@AndyAyersMS Alright, after yet more thinking, `gtGetParent` is actually 100% safe and does what is expected here (returns `nullptr` for the "no parent" case). In light of this, those are the outcomes of calling it here as I see them:

1\) We get a `GT_COMMA` parent.
- The check is the first operand - proceed to remove it.
- The check is the second operand - this should be invalid IR, as the check does not produce a value. Is that correct?

2\) We get non-`GT_COMMA`, non-`nullptr` parent. Is that also invalid IR?

3\) We get `nullptr` - we have a top-level check.

The checks here seem quite defensive - it appears to me that if the cases that are "invalid IR" above really are invalid, all of them could be dropped (turned into asserts).

So my question is then, are they and even if they are, should we still guard?

Reply:
Agree it seems like case 1a and 2 should never come up. So adding asserts for those seems like a good idea. But probably still prudent to guard.

You might be able to use `noway_asserts` to avoid redundant work, but since the redundancy is only for chk builds I'm not sure I would worry that much.